google-cloud-storage/test/google/cloud/storage/retry/utils.rb

```diff
@@ -0,0 +1,462 @@
+    "storage.notifications.delete" => [:delete_notification],
+    "storage.notifications.get" => [
+      :get_notification,
+      :bucket_get_notification
+    ],
+    "storage.notifications.list" => [
+      :list_notifications,
+      :bucket_notifications
+    ],
+    "storage.object_acl.list" => [:list_file_acls],
+    "storage.object_acl.delete" => [:delete_file_acl],
+    "storage.object_acl.insert" => [:insert_file_acl],
+    "storage.objects.get" => [
+      :get_object,
+      :get_file,
+      :download_file
+    ],
+    "storage.objects.insert" => [
+      :insert_object
+    ],
+    "storage.objects.compose" => [:compose_file],
+    "storage.objects.delete" => [:delete_file],
+    "storage.objects.list" => [:list_files],
+    "storage.objects.patch" => [:patch_file],
+    "storage.objects.rewrite" => [:rewrite_file],
+    "storage.serviceaccount.get" => [:project_service_account],
+    "storage.buckets.patch" => [
+      :patch_bucket,
+      :bucket_acl_public,
+      :bucket_acl_private,
+      :default_acl_owner_full,
+    ],
+    "storage.buckets.setIamPolicy" => [:set_bucket_policy],
+    "storage.hmacKey.update" => [:update_hmac_key],
+    "storage.resumable.upload" => [
+      :blob_upload_from_string,
+      :blob_upload_from_file,
+      :blob_upload_from_filename,
+      :blobwriter_write,
+    ],
+    "storage.notifications.insert" => [
+      :insert_notification,
+      :create_notification
+    ],
+    "storage.bucket_acl.delete" => [
+      :delete_bucket_acl,
+      :delete_acl
+    ],
+    "storage.bucket_acl.insert" => [
+      :insert_bucket_acl,
+      :insert_acl
+    ],
+    "storage.default_object_acl.delete" => [
+      :delete_bucket_default_acl,
+      :delete_default_acl
+    ],
+    "storage.default_object_acl.insert" => [
+      :insert_bucket_default_acl,
+      :insert_default_acl
+    ]
+  }
+  end
+
+  ######################################################################################
+  ### Library methods for mapping ######################################################
+  ######################################################################################
+
+  def self.list_bucket_acls client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    client.list_bucket_acls bucket_name
+  end
+
+  def self.delete_bucket client, _preconditions, **resources
+    bucket = resources[:bucket]
+    bucket.files.all { |f| f.delete rescue nil }
+    bucket.delete
+  end
+
+  def self.get_bucket client, _preconditions, **resources
+    client.get_bucket resources[:bucket].name
+  end
+
+  def self.bucket_reload client, _preconditions, **resources
+    bucket = resources[:bucket]
+    bucket.reload!
+  end
+
+  def self.get_bucket_policy client, _preconditions, **resources
+    bucket_name = resources[:bucket].name
+    client.get_bucket_policy bucket_name
+  end
+
+  def self.insert_bucket client, _preconditions, **resources
+    bucket_name = "new_bucket" + Time.now.to_i.to_s
```

Comment:
How do we avoid conflicts / duplicate bucket names if multiple tests are running at the same time? This can get flaky.

Reply:
Agreed. It's best to append some random digits to prevent this.

Reply:
Yes. random upto 10000 should be a safer bet